pkg/exprhelpers/helpers.go

```diff
@@ -130,20 +130,26 @@ func UpdateRegexpCacheMetrics() {
 
 func FileInit(fileFolder string, filename string, fileType string) error {
 	log.Debugf("init (folder:%s) (file:%s) (type:%s)", fileFolder, filename, fileType)
+	if fileType == "" {
+		log.Debugf("ignored file %s%s because no type specified", fileFolder, filename)
+		return nil
+	}
+	ok, err := existsInFileMaps(filename, fileType)
+	if ok {
+		log.Debugf("ignored file %s%s because already loaded", fileFolder, filename)
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
 	filepath := filepath.Join(fileFolder, filename)
 	file, err := os.Open(filepath)
 	if err != nil {
 		return err
 	}
 	defer file.Close()
 
-	if fileType == "" {
-		log.Debugf("ignored file %s%s because no type specified", fileFolder, filename)
-		return nil
-	}
-	if _, ok := dataFile[filename]; !ok {
-		dataFile[filename] = []string{}
```

Comment:
I may miss something, but I don't see where the global variable dataFile is allocated ?

Reply:
They were already being made here https://github.com/crowdsecurity/crowdsec/blob/0fc68d8490a075ece3e7a7d0f1980c87ad6a2596/pkg/exprhelpers/helpers.go#L35-L37